pyk/src/pyk/kast.py

```diff
@@ -1223,6 +1223,26 @@ def __iter__(self) -> Iterator[KSentence]:
     def productions(self) -> List[KProduction]:
         return [sentence for sentence in self.sentences if type(sentence) is KProduction]
 
+    @property
+    def syntax_productions(self) -> List[KProduction]:
+        return [prod for prod in self.productions if prod.klabel]
+
+    @staticmethod
+    def _is_non_free_constructor(label: str) -> bool:
```

Comment:
I moved it into the `functions` property, because I'm hoping that there are not many places where this distintion between injective functions and constructors is important. If it does become important, we should revisit this issue: https://github.com/runtimeverification/k/issues/2683